Amethyst/Managers/WindowManager.swift

```diff
@@ -620,6 +643,17 @@ final class WindowManager: NSObject, MouseStateKeeperDelegate {
 
             if screenManager == nil {
                 screenManager = ScreenManager(screen: screen, screenIdentifier: screenIdentifier, delegate: self, userConfiguration: userConfiguration)
+                screenManager!.onReflowInitiation = {
+                    self.mouseStateKeeper.handleReflowEvent()
+                }
+                screenManager!.onReflowCompletion = {
+                    // This handler will be executed by the Operation, in a queue.  Although async
```

Comment:
I think you should probably ensure that this happens on the main queue. If you're worried about ordering of completions you could compare the dates to make sure we only ever take the most recent available date.

Reply:
I'm not worried about the ordering, I just wanted to address the fact that I'm not providing any thread safety on an assignment operation that could happen in multiple threads.  Since the value is always increasing, and since the fudge factor is in tenths of a second, I think we're safe here.

Reply:
Also, I'm not sure what's meant by "ensure that this happens on the main queue"